### PR TITLE
Update plugin install instructions for OpenSearch

### LINKED ISSUE #25: Update install instructions for plugins
The instructions to install a plugin in OpenSearch also need a reference to publishing dependencies to Maven local. There are some typos and broken link references in the instructions.

Raising this issue to fix the above.

### PR DESCRIPTION
**Pull request by VachaShah**
### Description
This change is for updating instructions, typo and links in the `BUILDING.md` and `INSTALLING.md`.
 
### Issues Resolved
#25 
 
### Check List
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: BUILDING.md [hunk 1]
```diff
@@ -1,9 +1,10 @@
-## Building Plugins with OpenSeearch
+## Building Plugins with OpenSearch
 
 Until OpenSearch and other artifacts are published to Maven Central [OpenSearch#581](https://github.com/opensearch-project/OpenSearch/issues/581), plugins may require building all their dependencies and publishing them to Maven local.
 
-- [Publish OpenSearch to Maven Local](#publish-opensearch-to-maven-local)
-- [Use OpenSearch from Maven Local in Plugins](#use-opensearch-from-maven-local-in-plugins)
+- [Building Plugins with OpenSearch](#building-plugins-with-opensearch)
```
Comment (saratvemulapalli) [inline BUILDING.md:5]: Do we need adding this section? as this content is already part of it?
If you'd like to have a tree, probably we can put this at the top just like https://github.com/opensearch-project/opensearch-plugins/blob/main/README.md

File: INSTALLING.md [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
-- [Installing OpenSearch Plugins](#installing-opensearch-plugins)
-- [Installing OpenSearch Dashboards Plugins](#installing-opensearch-dashboards-plugins)
+- [Installing Plugins](#installing-plugins)
+  - [Installing OpenSearch Plugins](#installing-opensearch-plugins)
+  - [Installing OpenSearch Dashboards Plugins](#installing-opensearch-dashboards-plugins)
 
 ### Installing OpenSearch Plugins
 
-Assemble, extract and run OpenSearch `1.0.0-beta1` using [Building OpenSearch package](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.asciidoc#creating-packages).  
+Assemble, extract and run OpenSearch `1.0.0-beta1` using [Building OpenSearch package](https://github.com/opensearch-project/OpenSearch/blob/main/TESTING.md#creating-packages).
```
Comment (saratvemulapalli) [inline INSTALLING.md:9]: Thanks for this. 

We should automatically find broken links, can you create an issue for this?
Quick googling: Github actions supports verifying broken links, https://github.com/marketplace/actions/broken-link-check. We could do something similar.

**Review by dblock — APPROVED** [review]

**Review by saratvemulapalli — COMMENTED** [review]
Overall the changes look good to me.

File: INSTALLING.md (hunk 2 above)
Comment (saratvemulapalli) [reply INSTALLING.md:9]: created an issue: https://github.com/opensearch-project/opensearch-plugins/issues/29

**Pull request merged**